mypy failing

mypy is failing in the tests with the following error

```
435: error: Incompatible return value type (got "Message", expected "PackageMetadata")  [return-value]
435: note: Following member(s) of "Message" have conflicts:
435: note:     Expected:
435: note:         def [_T] get_all(self, name: str, failobj: _T = ...) -> Union[List[Any], _T]
435: note:     Got:
435: note:         @overload
435: note:         def get_all(self, name: str, failobj: None = ...) -> Optional[List[Any]]
435: note:         @overload
435: note:         def [_T] get_all(self, name: str, failobj: _T) -> Union[List[Any], _T]
```

The typeshed definitions changed in https://github.com/python/typeshed/commit/b5a26d1ca258a86a6f133dfabf58790553dc21cf, causing `PackageMetadata` to fall out of sync. I think the right option here is just to bring `PackageMetadata` back in sync with `email.message.Message`

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,3 +1,10 @@
+v6.0.1
+======
+
+* #434: Expand protocol for ``PackageMetadata.get_all`` to match
+  the upstream implementation of ``email.message.Message.get_all``
+  in python/typeshed#9620.
+
 v6.0.0
 ======
 

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -61,9 +61,11 @@ intersphinx_mapping.update(
     ),
 )
 
-# Workaround for #316
 nitpick_ignore = [
+    # Workaround for #316
     ('py:class', 'importlib_metadata.EntryPoints'),
     ('py:class', 'importlib_metadata.SelectableGroups'),
     ('py:class', 'importlib_metadata._meta._T'),
+    # Workaround for #435
+    ('py:class', '_T'),
 ]

--- a/importlib_metadata/_meta.py
+++ b/importlib_metadata/_meta.py
@@ -21,7 +21,7 @@ class PackageMetadata(Protocol):
     # overload per python/importlib_metadata#435
     @overload
     def get_all(self, name: str, failobj: None = None) -> Optional[List[Any]]:
-        ...
+        ...  # pragma: no cover
 
     @overload
     def get_all(self, name: str, failobj: _T) -> Union[List[Any], _T]:

--- a/importlib_metadata/_meta.py
+++ b/importlib_metadata/_meta.py
@@ -18,6 +18,7 @@ class PackageMetadata(Protocol):
     def __iter__(self) -> Iterator[str]:
         ...  # pragma: no cover
 
+    # overload per python/importlib_metadata#435
     @overload
     def get_all(self, name: str, failobj: None = None) -> Optional[List[Any]]:
         ...

--- a/importlib_metadata/_meta.py
+++ b/importlib_metadata/_meta.py
@@ -1,5 +1,5 @@
 from ._compat import Protocol
-from typing import Any, Dict, Iterator, List, TypeVar, Union
+from typing import Any, Dict, Iterator, List, Optional, TypeVar, Union, overload
 
 
 _T = TypeVar("_T")
@@ -18,7 +18,12 @@ class PackageMetadata(Protocol):
     def __iter__(self) -> Iterator[str]:
         ...  # pragma: no cover
 
-    def get_all(self, name: str, failobj: _T = ...) -> Union[List[Any], _T]:
+    @overload
+    def get_all(self, name: str, failobj: None = None) -> Optional[List[Any]]:
+        ...
+
+    @overload
+    def get_all(self, name: str, failobj: _T) -> Union[List[Any], _T]:
         """
         Return all values associated with a possibly multi-valued key.
         """

